internal/store/state.go

```diff
@@ -0,0 +1,20 @@
+package store
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+)
+
+// NewState generates cryptographically secure random state with base64 URL encoding
+func NewState() (string, error) {
+	cryptoRandBytes := make([]byte, 16) //nolint:gomnd
+	_, err := rand.Read(cryptoRandBytes)
+
+	if err != nil {
+		return "", err
+	}
```

Comment:
nit: cuddle the err check with the definition 
```suggestion
	cryptoRandBytes := make([]byte, 16) //nolint:gomnd
	
	_, err := rand.Read(cryptoRandBytes)
	if err != nil {
		return "", err
	}
```